strapi/strapi
# fix: Private s3 bucket with CDN & compatible S3 providers

### What does it do?

When using the private configuration in the s3 provider, we parse assets url to find if they are from the configured provider bucket.

Parsing did not work for some compatible S3 providers ( DO spaces, and others) nor for CDN's.

CDN's are tricky, the ones we have tried (cloudfront) require a different package to sign the urls. The workaround is to keep using the same aws sdk package, and return the signed private bucket url not the signed cdn url.

Returning signed cdn urls would require user customization of the `getSignedUrl` method.

### Why is it needed?

Have a private S3 Provider configuration when using CDN's or other compatible S3 providers 

### How to test it?

- See tests
- Ensure there is nothing broken for image signing

### Related issue(s)/PR(s)
https://github.com/strapi/strapi/issues/16245
